scripts/installer_test.go

```diff
@@ -0,0 +1,339 @@
+// See LICENSE for copying information.
+
+// +build windows
+// +build ignore
+
+package main
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zeebo/errs"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"storj.io/storj/private/testcontext"
+)
+
+var (
+	releaseMSIPath string
+	// TODO: make this more dynamic and/or use versioncontrol server?
+	// (NB: can't use versioncontrol server until updater process is added to response)
+	oldRelease  = "v0.26.2"
+	msiBaseArgs = []string{
+		"/quiet", "/qn",
+		"/norestart",
+	}
+
+	msiPathFlag = flag.String("msi", "", "path to the msi to use")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *msiPathFlag == "" {
+		log.Fatal("no msi passed, use `go test ... -args -msi \"<msi path (using \\ seperator)>\"`")
+	}
+
+	tmp, err := ioutil.TempDir("", "release")
+	if err != nil {
+		panic(err)
+	}
+
+	releaseMSIPath = filepath.Join(tmp, oldRelease+".msi")
+	if err = downloadInstaller(oldRelease, releaseMSIPath); err != nil {
+		panic(err)
+	}
+	defer func() {
+		err = errs.Combine(os.RemoveAll(tmp))
+	}()
+
+	os.Exit(m.Run())
+}
+
+func TestInstaller_Config(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	installDir := ctx.Dir("install")
+	configPath := ctx.File("install", "config.yaml")
+
+	walletAddr := "0x0000000000000000000000000000000000000000"
+	email := "user@mail.test"
+	publicAddr := "127.0.0.1:10000"
+
+	args := []string{
+		fmt.Sprintf("STORJ_WALLET=%s", walletAddr),
+		fmt.Sprintf("STORJ_EMAIL=%s", email),
+		fmt.Sprintf("STORJ_PUBLIC_ADDRESS=%s", publicAddr),
+	}
+	install(t, ctx, *msiPathFlag, installDir, args...)
+	defer requireUninstall(t, ctx)
+
+	expectedEmail := fmt.Sprintf("operator.email: %s", email)
+	expectedWallet := fmt.Sprintf("operator.wallet: \"%s\"", walletAddr)
+	expectedAddr := fmt.Sprintf("contact.external-address: %s", publicAddr)
+
+	configStr := readConfigLines(t, ctx, configPath)
+
+	require.Contains(t, configStr, expectedEmail)
+	require.Contains(t, configStr, expectedWallet)
+	require.Contains(t, configStr, expectedAddr)
+}
+
+func TestUpgrade_Config(t *testing.T) {
+	if *msiPathFlag == "" {
+		t.Fatal("no msi passed, use `go test ... -args -msi <msi path>`")
+	}
+
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	installDir := ctx.Dir("install")
+	configPath := ctx.File("install", "config.yaml")
+
+	walletAddr := "0x0000000000000000000000000000000000000000"
+	email := "user@mail.test"
+	publicAddr := "127.0.0.1:10000"
+
+	// install old release
+	args := []string{
+		fmt.Sprintf("STORJ_WALLET=%s", walletAddr),
+		fmt.Sprintf("STORJ_EMAIL=%s", email),
+		fmt.Sprintf("STORJ_PUBLIC_ADDRESS=%s", publicAddr),
+	}
```

Comment:
The installer does not collect any user data in upgrade mode, so we shouldn't set any of these.

Reply:
Oh, now I see this is for the initial installation that will be upgraded... Forget what I said...